### DISCUSSION TITLE
项目打包运行问题

### DISCUSSION THREAD

**Discussion by 1243excz** [galacean/discussion#119] [Editor] [answered]
![Snipaste_2024-04-15_09-52-50](https://github.com/galacean/discussion/assets/167043713/272731a2-d774-4a4f-8d25-9ad0482d0e78)
GITF Meshopt取消勾选，其他默认。Platform 选择React，npm run build之后dist里面没有html。选择Vanilla之后，npm run build将dist里面的内容传到oss。显示找不到文件路径。在head头里面加上base标签指定路径然后在vite.config.js里面加上  base: './'。然后出现另一个bug：打包完成之后，html这里显示是小写的assets：<script type="module" crossorigin src="./assets/index-f0d308fc.js"></script>,但是dist下面是大写的Assets文件夹。导致文件找不到。手动修改之后，又报了另一个错：Failed to load resource: the server responded with a status of 404 ()
index-f0d308fc.js:4173  Failed to start project Error: request failed from: /project.json
    at r.onload (index-f0d308fc.js:1373:71674)

这要怎么改？

**Comment by luzhuang** [comment]
@gz65555 看看

**Comment by gz65555** [comment]
@1243excz 项目的链接发一下

**Reply by 1243excz** [reply]
@gz65555 
https://resources.test.easy-h5.cn/LSY/goo2/index.html

**Reply by gz65555** [reply]
@1243excz 你的编辑器的项目链接，我这边导出测试一下，看看具体是什么问题😂

**Reply by 1243excz** [reply]
https://galacean.antgroup.com/editor/#/project/39400003   @gz65555 感谢！

**Reply by gz65555** [reply]
我测试了一下，不需要改任何构建配置，只需要把 serve 的 root path 设置在 dist 下面。

**Reply by gz65555** [reply]
oss 不适合部署网页，只适合放 public 下面的静态资源

**Reply by 1243excz** [reply]
据我了解，oss是可以部署静态页面。我们的项目需求是不部署在根目录下，类似于github 的page功能。这种情况要怎么处理？

**Reply by gz65555** [reply, marked as answer]
这个目前在编辑器里还没有去支持特定的 base。目前你可以手动去修改下根目录下的 project.json 的 url 字段。还有 public 里面的 project.json 文件，把 assets[i].path 字段。把绝对路径设置成相对。

**Reply by 1243excz** [reply]
@gz65555 我看到项目中的路径都是 /xxx/xx 从根目录开始。vite.config.js里面加上 base: '/LSY/xx'，我自己的路径，但是一些资源还是路径不对。有什么快速的方法可以前面拼接上我自己的路径吗？

**Reply by 1243excz** [reply]
@gz65555 他现在是在根目录下面查找。但是我想在当前目录下面查找，编辑器中有可以设置路径的吗

**Reply by gz65555** [reply]
第一个问题，我也不是特别清楚怎么快速加上路径，按我的理解，你把路径的 `/` 改成 `./` 就行。 
第二个问题，编辑器目前还没有这个功能，不过你的建议不错，可以考虑加上。

**Comment by 1243excz** [comment]
@gz65555 一定要尽快加上呀，哥！手动调路径是真的有点麻烦。

**Reply by gz65555** [reply]
已经记录 issue，后续会进入排期
